### ISSUE TITLE
IdGraph: Not all loop groups get promotion domains

### ISSUE THREAD

**Issue by naoyam** [NVIDIA/Fuser#1146]
If no inline promotion happens, no loop promotion is set, which means some loop groups have no promotion domain. That is a little inconvenient since the loop promotion will be used as the concrete ID in the current CompuAtMap graph.

**Comment by naoyam** [comment]
For example:

```
[ RUN      ] NVFuserTest.FusionAdvancedComputeAt7_CUDA
unknown file: Failure
C++ exception with description "idGraph(IdMappingMode::EXACT).toGroup(promotion) == idGraph(IdMappingMode::EXACT).toGroup(existing_promotion) INTERNAL ASSERT FAILED at "/home/nmaruyama/nvfuser/debug1/csrc/id_model/id_graphs.cpp":1831, p
lease report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Different promotions found for idg{11 13 18 20 46 48 49 50 53 55 57}. iS13{( ( (( (( getMetaData(T0) )).logical_size ))[0] ) * ( ceilDiv(( (( (( getMetaData(T6) )).logical_size ))[1] ), 2) ) )}, iS55{( ceilDiv(( (( (( getMetaData(T6) )).logical_size ))[1] ), 2) )}
Exception raised from operator() at /home/nmaruyama/nvfuser/debug1/csrc/id_model/id_graphs.cpp:1831 (most recent call first):
frame #0: nvfuser::nvfCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x7a (0x7f1a88b2329a in /home/nmaruyama/nvfuser/debug1/build/libnvfuser_codegen.so)
frame #1: nvfuser::nvfErrorFail(char const*, char const*, unsigned int, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x5d (0x7f1a88b2350d in /home/nmaruyama/nvfuser/debug1/build/libnvfuser_codegen.so)
frame #2: <unknown function> + 0x15532fa (0x7f1a88c4d2fa in /home/nmaruyama/nvfuser/debug1/build/libnvfuser_codegen.so)
frame #3: <unknown function> + 0x1550af5 (0x7f1a88c4aaf5 in /home/nmaruyama/nvfuser/debug1/build/libnvfuser_codegen.so)
frame #4: nvfuser::IterDomainGraphs::build(std::vector<nvfuser::Expr*, std::allocator<nvfuser::Expr*> > const&, std::vector<nvfuser::TensorView*, std::allocator<nvfuser::TensorView*> > const&) + 0x5c1 (0x7f1a88c3eca1 in /home/nmaruyama/nvfuser/debug1/build/libnvfuser_codegen.so)
```

**Comment by naoyam** [comment]
Even with #1149, loop groups appearing before forwarded domains may still not have unique promotions. I'd argue that those groups themselves do not make much sense, however. I'll ignore them for now.

**Comment by naoyam** [comment]
Closed with #1149
